src/PageHeader/manageResponsiveValueWithCSS.ts

```diff
@@ -0,0 +1,130 @@
+import {isResponsiveValue, ResponsiveValue} from '../hooks/useResponsiveValue'
+import {BetterSystemStyleObject} from '../sx'
+
+function areAllValuesTheSame(responsiveValue: ResponsiveValue<boolean | number | string>): boolean {
+  if ('narrow' in responsiveValue && 'regular' in responsiveValue && 'wide' in responsiveValue) {
+    const values = Object.values(responsiveValue)
+    return values.every(value => value === values[0])
+  }
+  return false
+}
+function haveRegularAndWideSameValue(responsiveValue: ResponsiveValue<boolean | number | string>): boolean {
+  if ('regular' in responsiveValue && 'wide' in responsiveValue) {
+    return responsiveValue.regular === responsiveValue.wide
+  }
+  return false
+}
+
+/**
+ * This function is inspired by the `useResponsiveValue` hook and it's used to render responsive values with CSS.
+ * @param value - The value that needs to be rendered responsively
+ * @param cssProperty - The CSS property whoes value needs to be rendered responsively
+ * @param comparisonFunc - A function that evaluates which value to assign to the CSS property
+ 
+ * @example
+ * CSSManagedResponsiveValue({narrow: true, regular: true, wide: false}, 'display', value => {
+      return value ? 'none' : 'flex'
+    })
+ * @returns
+ * {
+ *   "@media screen and (max-width: 768px)": {
+ *     "display": "none"
+ *   },
+ *   "@media screen and (min-width: 768px)": {
+ *     "display": "none"
+ *   },
+ *   "@media screen and (min-width: 1440px)": {
+ *     "display": "flex"
+ *   }
+ * }
+ * 
+ * * @example
+ * CSSManagedResponsiveValue({regular: 'border.default', wide: 'canvas.inset'}, 'backgroundColor', (value): string => {
+    return value
+  })
+ * @returns
+ * {
+ *   "@media screen and (min-width: 768px)": {
+ *     "backgroundColor": "border.default"
+ *   },
+ *   "@media screen and (min-width: 1440px)": {
+ *     "backgroundColor": "canvas.inset"
+ *   }
+ * }
+ * 
+ * * @example
+* CSSManagedResponsiveValue({narrow: 'filled', regular: 'line'}, 'height', (value): string => {
+    return {
+      filled: 8,
+      line: 1,
+    }[value]
+  })
+ * @returns
+ * {
+ *  "@media screen and (max-width: 768px)": {
+ *     "height": 8
+ *   }
+ *   "@media screen and (min-width: 768px)": {
+ *     "height": 1
+ *   },
+ * }
+ */
+export function CSSManagedResponsiveValue<T>(
+  value: T,
+  cssProperty: string,
+  comparisonFunc: (value: T) => void,
```

Comment:
A couple of notes:
- `comparisonFunc` feels like the wrong name since this function is transforming one value into another value, not comparing two values. I think `transformFn` or `mapFn` would be a more appropriate name.
- We use the return value of this function, so it's return type shouldn't be `void`. I think the correct type signature might look something like this:

```js
import * as CSS from "csstype";

function responsiveCssProperty<TInput, TOutput>(
  cssProperty: keyof CSS.Properties,
  mapFn: (value: TInput) => TOutput,
  value: TInput | ResponsiveValue<TInput>
) { ...
```

Reply:
Definitely love the `mapFn` 💖  Josh also suggested that. And sorry about the types, I completely forgot to update them. Love the way you typed them 💖